pkg/virt-handler/virtwrap/manager.go

```diff
@@ -322,35 +323,70 @@ func NewLibvirtDomainManager(connection Connection, recorder record.EventRecorde
 	return &manager, nil
 }
 
+// Check if a Domain with the VM name and UID exists. If a Domain with the VM name but with a different
+// UID exists, delete it. If everything matches, return the loaded Domain.
+func (l *LibvirtDomainManager) getActualOrDeleteOutdatedDomain(vm *v1.VM) (VirDomain, bool, error) {
+	// In case it does not exist, check if another VM with that name exists already
+	dom, err := l.virConn.LookupDomainByName(vm.ObjectMeta.GetName())
+	exists := err == nil
+	connectionErr := err != nil && err.(libvirt.Error).Code != libvirt.ERR_NO_DOMAIN
+	if connectionErr {
+		logging.DefaultLogger().Object(vm).Error().Reason(err).Msg("Getting the domain failed.")
+		return nil, false, err
+	}
+	if (!exists) {
+		return nil, false, nil
+	}
+	uid, err := dom.GetUUIDString()
+	if err != nil {
+		dom.Free()
+		logging.DefaultLogger().Object(vm).Error().Reason(err).Msg("Getting the UID of the domain failed.")
+		return nil, false, err
+	}
+	// Check if the UIDs match and delete the old VM if they don't
+	if string(vm.GetObjectMeta().GetUID()) != uid {
+		vm := v1.NewVMReferenceFromName(vm.ObjectMeta.Name)
+		vm.GetObjectMeta().SetUID(types.UID(uid))
+		if err = l.killDomain(vm, dom); err != nil {
```

Comment:
Hm, I would have to do a complete copy of the cache.sharedIndexerInformer. I can't replace the `DeletionHandlingMetaNamespaceKeyFunc` otherwise ...